storage/imxrt-flash/flashsrv.c

```diff
@@ -982,8 +997,25 @@ static int flashsrv_customInit(void)
 }
 
 
-int main(void)
+int main(int argc, char **argv)
 {
+	for (;;) {
+		int c = getopt(argc, argv, "o");
+		if (c < 0) {
+			break;
+		}
+
+		switch (c) {
+			case 'o': /* Allows to restrict operations on raw partitions to process that opens partition first */
```

Comment:
Better mechanism would be if you could pass partition(s) to restrict, instead of enable/disable restriction globally.

Reply:
Other ideas I've just thought of:
 - ditch the commandline param, abuse `O_EXCL` mode of `open` - if set - allow only this PID until `close` (maaaybe can even be used for meterfs (but not for every file open, just for the partition file?))
 - ditch the commandline param, use POSIX locking mechanism API (`fcntl` + `F_SETLK`) - probably harder :)

Reply:
> * ditch the commandline param, abuse `O_EXCL` mode of `open` - if set - allow only this PID until `close` (maaaybe can even be used for meterfs (but not for every file open, just for the partition file?))

I like this approach. I figured that maybe exclusive access could be granted only if `O_CREAT` flag is not set as behaviour with `O_CREAT | O_EXCL` is well defined and for `O_EXCL` is not.